base/event.jl

```diff
@@ -5,33 +5,114 @@
 
 if JULIA_PARTR
 
+# Global TODO: arg and error
+
 import Core.Condition
 
+"""
+    Condition()
+
+Create an edge-triggered event source that tasks can wait for. Tasks that
+call [`wait`](@ref) on a `Condition` are suspended. They are woken up when
+[`notify`](@ref) is later called on the `Condition`. Edge triggering means
+that only tasks waiting at the time [`notify`](@ref) is called are woken
+up. For level-triggered notifications, you must keep extra state to keep
+track of whether a notification has happened. The [`Channel`](@ref) type
+does this, and so can be used for level-triggered events.
+"""
 Condition() = ccall(:jl_condition_new, Ref{Condition}, ())
+
+"""
+    wait([x])
```

Comment:
I think we want to attach this docstring (since it is so general) to `function wait end` and not `wait(condition)`.
IIUC docstrings get accumulated over methods and shown when asked for the generic function one, but you can also ask for a specific docstring once you have a method